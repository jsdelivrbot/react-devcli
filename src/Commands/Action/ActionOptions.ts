export const name = 'action';

export const description = 'Action description';

export const flags = {
    component: {
        alias: 'c',
        demand: false,
        describe: 'Create component alongside action file',
    }
};
