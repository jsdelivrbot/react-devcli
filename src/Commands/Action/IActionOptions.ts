export interface IActionOptions {
    name: string;
    actions: string[];
    separateActionTypes?: true;
    reducer?: boolean;
    selector?: boolean;
}
