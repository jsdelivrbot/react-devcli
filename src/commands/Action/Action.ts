import {Arguments} from 'yargs';
import ACommand from '../Command/ACommand';
import {IMakeable} from '../Command/contracts/IMakeable';
import {IRemovable} from '../Command/contracts/IRemovable';
import {ITestable} from '../Command/contracts/ITestable';
import {ActionsConfig} from 'react-devcli';

class Action extends ACommand implements IRemovable, ITestable, IMakeable {
    constructor() {
        const name = 'action';
        const description = 'Action description';
        const options = {
            component: {
                alias: 'c',
                demand: false,
                describe: 'Create component alongside action file',
            },
        };

        super(name, description, options);
    }

    public run(argv: Arguments, config: ActionsConfig): void {
        console.log(argv);
        console.log(config);
    }

    public help(argv: Arguments): void {
        console.log(argv);
    }

    public remove(argv: Arguments): void {
        console.log(argv);
    }

    public test(argv: Arguments): void {
        console.log(argv);
    }

    public make(argv: Arguments): void {
        console.log(argv);
    }
}

export default new Action();
