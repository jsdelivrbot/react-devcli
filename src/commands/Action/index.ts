import Action from './Action';

export default new Action();
