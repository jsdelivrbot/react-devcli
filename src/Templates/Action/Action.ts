import {findAndReplace} from '../../Utils/Template/Template';
import {IActionOptions} from '../../Commands/Action/IActionOptions';

export const js = (actionsConfig: IActionOptions): string => {
    const {actions, name, separateActionTypes} = actionsConfig;
    const actionTypeFileName = `${name[0].toUpperCase() + name.split('').slice(1).join('')}Actions`;
    const actionTypesConstants = actions.map((a): string => a.toUpperCase());
    const actionTypesImports = `import {${actionTypesConstants.join(', ')}} from './${actionTypeFileName}';`;
    const actionTypesExports = actionTypesConstants
        .map((actionType): string => `export const ${actionType} = '${actionType}';`)
        .join('\n');

    const actionExport = [
        '',
        'export const REACT_ACTION_NAME = () => ({',
        '    type: REACT_ACTION_TYPE',
        '});',
        ''
    ].join('\n');

    const actionExports = actions
        .map((a): string => {
            return findAndReplace(actionExport,
                {
                    REACT_ACTION_NAME: a,
                    REACT_ACTION_TYPE: a.toUpperCase()
                }
            );
        })
        .join('');

    return [
        separateActionTypes && actionTypesImports,
        !separateActionTypes && actionTypesExports,
        actionExports,
        ''
    ].filter((e): any => e).join('\n');
};

export const ts = (): string => {
    return '';
};
