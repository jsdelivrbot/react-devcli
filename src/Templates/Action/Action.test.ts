import {js, ts} from './Action';

describe('Action template test suite', (): void => {
    test('should match action template snapshot for JS language', (): void => {
        expect(js()).toMatchSnapshot();
    });

    test('should match action template snapshot for TS language', (): void => {
        expect(ts()).toMatchSnapshot();
    });
});
