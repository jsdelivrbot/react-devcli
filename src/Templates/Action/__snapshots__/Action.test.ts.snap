// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action template test suite should match action template snapshot for JS language 1`] = `""`;

exports[`Action template test suite should match action template snapshot for TS language 1`] = `""`;
